projects/packages/connection/legacy/class-jetpack-xmlrpc-server.php

```diff
@@ -61,6 +61,7 @@ public function xmlrpc_methods( $core_methods ) {
 			'jetpack.remoteProvision'  => array( $this, 'remote_provision' ),
 			'jetpack.idcUrlValidation' => array( $this, 'validate_urls_for_idc_mitigation' ),
 			'jetpack.unlinkUser'       => array( $this, 'unlink_user' ),
+			'jetpack.disconnectBlog'   => array( $this, 'disconnect_blog' ),
 		);
 
 		if ( class_exists( 'Jetpack' ) ) {
```

Comment:
It looks like the `disconnect_blog()` method requires Jetpack-the-plugin.
Should we move it into under the `class_exists( 'Jetpack' )` condition?
```suggestion
		if ( class_exists( 'Jetpack' ) ) {
			$jetpack_methods['jetpack.disconnectBlog']    = array( $this, 'disconnect_blog' );
```

Reply:
@sergeymitr Hi Sergey! Thanks for reviewing this!

TBH I noticed that, but thought that maybe it makes more sense instead of moving it under the `class_exists( 'Jetpack' )` to make it not actually require Jetpack-the-plugin.

Nevertheless, this seemed out of the scope of the current PR so maybe we could discuss and address it in an other PR?

Reply:
Hi @fgiannar!

> to make it not actually require Jetpack-the-plugin

That's definitely the better solution, but I agree that it's out of scope right now.

However, I believe that we should still put it under the `class_exists( 'Jetpack' )` check. Decoupling the endpoint from Jetpack may take a long time, and we shouldn't make the endpoint throw a fatal error in the meantime.

Instead, let's only add the endpoint if we know it'll work.

Reply:
@sergeymitr Sure and done! 👍